charms/kfp-api/tests/integration/test_charm.py

```diff
@@ -51,24 +54,9 @@ async def test_build_and_deploy(self, ops_test: OpsTest):
         )
         await ops_test.model.deploy(entity_url="kfp-viz", channel="2.0/stable", trust=True)
 
-        # deploy mysql-k8s charm
-        await ops_test.model.deploy(
-            "mysql-k8s",
-            channel="8.0/stable",
-            series="jammy",
-            constraints="mem=250M",
-            trust=True,
-        )
-        await ops_test.model.wait_for_idle(
-            apps=["mysql-k8s"],
-            status="active",
-            raise_on_blocked=True,
-            timeout=60 * 10,
-            idle_period=60,
-        )
-
+        # FIXME: This assertion belongs to unit tests
         # test no database relation, charm should be in blocked state
-        assert ops_test.model.applications[APP_NAME].units[0].workload_status == "blocked"
+        # assert ops_test.model.applications[APP_NAME].units[0].workload_status == "blocked"
```

Comment:
I disagree on this one. We need to test blocked state in integration tests when no relations are established. Unit test does not cover the full deployment and all events that happen during actual charm deployment, eg. update status event coming in before relation is established.

Reply:
I'm not sure.  I think we could simulate that scenario in unit tests.  Could establish a relation, run an update status, and then add data to the relation.

Reply:
I understand this assertion's only goal is to check the `kfp-api` unit goes to `BlockedStatus` when there is no database relation. Is there anything that we need from a deployed charm to check that?

Reply:
At this point we do not need anything from mysql charm, but while we wait for that relation in blocked state other event might come by (an we had an issue with update-status) that can affect state machine. We should simulate it in unit test, I guess that could cover it.